### PR TITLE
PADV-509: Add CertPREP Theme

### PR DESCRIPTION
**Pull request by sergivalero20**
## Ticket

https://agile-jira.pearson.com/browse/PADV-509

## Description

This PR adds the templates and assets for CertPREP theme on edx-platform.

## Testing:

- Enable comprehensive theming.
- Run the LMS: `make dev.up.lms`.
- Set `pearson-vue-theme` on a site.
- Compile the CertPREP theme.
- Verify that the CertPREP theme is applied correctly on Olive.
- Enable Course Operations.
- Test course operations features.

(For more information enter here: [How to enable themes)](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/changing_appearance/theming/enable_themes.html#enable-themes)

This site is involved in Course Operation [(Course Operation features)](https://github.com/Pearson-Advance/course_operations/blob/master/pearson_course_operation/docs/course_operations_features_juniper.md).

## Screenshots

Log In:
Log in

Dashboard:
dashboard

## Reviewers

- [ ] @Squirrel18 
- [ ] @anfbermudezme

### CODE REVIEW

**Comment by anfbermudezme** [comment]
Could you add in the PR test instructions the features related to course operations?

File: edx-platform/pearson-vue-theme/lms/templates/ccx/course_name.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<%page expression_filter="h"/>
+<%!
+  from django.utils.translation import ugettext as _
+  from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+  from openedx.core.djangolib.js_utils import (
+    dump_js_escaped_json, js_escaped_string
+  )
+%>
+
+<%ALTERNAVITE_STRINGS = configuration_helpers.get_value('ALTERNAVITE_STRINGS', {})%>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/ccx/course_name.html]: Are we using "ALTERNAVITE_STRINGS" in prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html [hunk 2]
```diff
@@ -8,16 +8,11 @@
 from django.utils.translation import ugettext as _
 %>
 % if len(course_date_blocks) > 0:
-    <h3 class="hd hd-6 handouts-header">${_("Upcoming Dates")}</h3>
+    <h3 class="hd hd-6 handouts-header">${_("Upcoming dates")}</h3>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html]: Did you check if this page wasn't replaced by and MFE? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/courseware/info.html [hunk 3]
```diff
@@ -0,0 +1,122 @@
+<%page expression_filter="h"/>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/courseware/info.html]: What are the custom changes to this template? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/header.html [hunk 4]
```diff
@@ -22,9 +22,19 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/header.html]: Is this part of the Olive template? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/header.html [hunk 5]
```diff
@@ -42,7 +52,7 @@
   </script>
 % endif
 
-<header class="global-header ${'slim' if course else ''}">
+<header class="global-header ${'slim' if course else ''} vue-header">
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/header.html:46]: Where are we using this class? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html [hunk 6]
```diff
@@ -14,6 +14,7 @@
   show_explore_courses = settings.FEATURES.get('COURSES_ARE_BROWSABLE')
   self.real_user = getattr(user, 'real_user', user)
   enable_help_link = settings.FEATURES.get('ENABLE_HELP_LINK')
+  show_programs_tab = configuration_helpers.get_value('SHOW_PROGRAMS_TAB', show_program_listing)
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html]: Are we using this setting in any of the prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html [hunk 7]
```diff
@@ -61,4 +62,4 @@
     % endif
     <%include file="user_dropdown.html"/>
   </div>
-</div>
+</div>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html]: Missing end of file line. @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html [hunk 8]
```diff
@@ -5,41 +5,22 @@
...
-        % if settings.LOGO_IMAGE_EXTRA_TEXT == 'edge':
-            <span class="font-italic"> | EDGE</span>
-        % endif
-    </a>
-  % endif
+  <a href="${configuration_helpers.get_value('LMS_ROOT_URL', branding_api.get_home_url())}">
+    <%block name="navigation_logo">
+    <img  class="logo" src="${branding_api.get_logo_url(is_secure)}" alt="${_("{platform_name} Home Page").format(platform_name=static.get_platform_name())}"/>
+    </%block>
+  </a>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html]: Is this part of the Olive template? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 9]
```diff
@@ -3,36 +3,35 @@
...
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
 %>
 
 <%
 ## This template should not use the target student's details when masquerading, see TNL-4895
 self.real_user = getattr(user, 'real_user', user)
 profile_image_url = get_profile_image_urls_for_user(self.real_user)['medium']
 username = self.real_user.username
+
+if configuration_helpers.get_value('SHOW_PROFILE_NAME_INSTEAD_OF_USERNAME', False):
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Are we using this setting in any of the prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 10]
```diff
@@ -43,17 +42,23 @@
         % if resume_block:
             <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${resume_block}" role="menuitem">${_("Resume your last course")}</a></div>
         % endif
-        % if not enterprise_customer_portal:
-            <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('dashboard')}" role="menuitem">${_("Dashboard")}</a></div>
-        % else:
-            <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${settings.ENTERPRISE_LEARNER_PORTAL_BASE_URL}/${enterprise_customer_portal.get('slug')}" role="menuitem">${_("Dashboard")}</a></div>
+        % if configuration_helpers.get_value('SHOW_DASHBOARD_IN_DROPDOWN', False):
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Are we using this setting in any of the prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 11]
```diff
@@ -43,17 +42,23 @@
...
+            <div class="mobile-nav-item dropdown-item dropdown-nav-item">
+                <a href="${reverse('dashboard')}" role="menuitem">${_("Dashboard")}</a>
+            </div>
+        % endif
+        % if configuration_helpers.get_value('SHOW_PROFILE_IN_DROPDOWN', False):
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Are we using this setting in any of the prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 12]
```diff
@@ -43,17 +42,23 @@
...
         % endif
-
-        <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('learner_profile', kwargs={'username': username})}" role="menuitem">${_("Profile")}</a></div>
         <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('account_settings')}" role="menuitem">${_("Account")}</a></div>
-        % if should_show_order_history:
+        % if should_redirect_to_order_history_microfrontend():
             <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${settings.ORDER_HISTORY_MICROFRONTEND_URL}" role="menuitem">${_("Order History")}</a></div>
         % endif
-        <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('logout')}" role="menuitem">${_("Sign Out")}</a></div>
+        % if configuration_helpers.get_value('EXTERNAL_SUPPORT_LINK', ''):
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Are we using this setting in any of the prod sites? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 13]
```diff
@@ -43,17 +42,23 @@
...
+        % if configuration_helpers.get_value('SHOW_PROFILE_IN_DROPDOWN', False):
+            <div class="mobile-nav-item dropdown-item dropdown-nav-item">
+                <a href="${reverse('learner_profile', kwargs={'username': username})}" role="menuitem">${_("Profile")}</a>
+            </div>
         % endif
-
-        <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('learner_profile', kwargs={'username': username})}" role="menuitem">${_("Profile")}</a></div>
         <div class="mobile-nav-item dropdown-item dropdown-nav-item"><a href="${reverse('account_settings')}" role="menuitem">${_("Account")}</a></div>
-        % if should_show_order_history:
+        % if should_redirect_to_order_history_microfrontend():
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Is this part of the Olive template? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/index.html [hunk 14]
```diff
@@ -2,70 +2,42 @@
...
+        iframe.src = iframe.src;
+        modal.style.display = "none";
+      }
+    </script>
+    <img src="${static.url('images/CertPREP_homepage_laptop.png')}" onclick="openVideoModal()">
+  </div>
+</div>
+<div class="video-modal" id="videoModal" onclick="closeVideoModal()">
+  <iframe id="videoModalIframe" src="https://player.vimeo.com/video/578134840?h=198d342005&title=0&byline=0&portrait=0">
+</div>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/index.html]: Missing end of file line. @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html [hunk 15]
```diff
@@ -16,9 +17,9 @@
   <div class="data-display-text" id="data-grade-config-text"></div>
   <br>
 
-  <p>${_("The CSV of anonymized student IDs download has been relocated to the Reports section below.")}</p>
-
- </div>
+  <p>${_("Click to download a CSV of anonymized student IDs:")}</p>
+  <p><input type="button" name="list-anon-ids" value="${_("Get Student Anonymized IDs CSV")}" data-csv="true" class="csv" data-endpoint="${ section_data['get_anon_ids_url'] }" class="${'is-disabled' if disable_buttons else ''}" aria-disabled="${'true' if disable_buttons else 'false'}" ></p>
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: Is this part of the Olive template? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html [hunk 16]
```diff
@@ -61,20 +64,22 @@ <h3 class="hd hd-3">${_("Reports")}</h3>
...
-    <div class="mb-15 problems">
+    <p>
+      <div class="problems">
         ${static.renderReact(
           component="ProblemBrowser",
-          id="react-problem-report",
+          id="react-block-listing",
           props={
             "courseId": course.id,
-            "excludeBlockTypes": ['html', 'video', 'discussion'],
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: Are these changes part of the Certprep changes? @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html [hunk 17]
```diff
@@ -61,20 +64,22 @@ <h3 class="hd hd-3">${_("Reports")}</h3>
...
-            "ShowBtnUi": "true",
+            "excludeBlockTypes": ['html', 'video', 'discussion']
           }
         )}
-    </div>
+      </div>
+    </p>
+
+    <p>
+      <input type="button" name="list-problem-responses-csv" value="${_("Download a CSV of problem responses")}" data-endpoint="${ section_data['get_problem_responses_url'] }" data-csv="true">
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: Is this part of the Olive template? @sergivalero20

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

**Comment by sergivalero20** [comment]
> Could you add in the PR test instructions the features related to course operations?

Done.

File: edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html (hunk 2 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html]: @Squirrel18  As far as I checked, the page replaced by an MFE was **course-home-fragment.html**. I left a comment in the ticket in Jira (https://agile-jira.pearson.com/browse/PADV-509). Anyway I can check for course_dates_fragment.html again.

File: edx-platform/pearson-vue-theme/lms/templates/courseware/info.html (hunk 3 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/courseware/info.html]: @Squirrel18  In this template only was changed some texts in several links and it was removed an element related to page-subtitle [commit](https://github.com/Pearson-Advance/openedx-themes/commit/181ab602d532a2a20d64dec68cd712205dbb4ba2#diff-6cc945d0e7aa824263069b6e9d6ab052a9f955160a53d08e02378303ebe311c6).

Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/courseware/info.html]: Info.html doesn't appear in Olive Version.

File: edx-platform/pearson-vue-theme/lms/templates/header/header.html (hunk 4 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/header.html]: @Squirrel18  No, it's just part of Juniper Version. I'll remove it.

File: edx-platform/pearson-vue-theme/lms/templates/header/header.html (hunk 5 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/header.html:46]: @Squirrel18  Only it's used to add an style in _navigation.scss (Juniper Version)

Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/header.html:46]: [commit](https://github.com/Pearson-Advance/openedx-themes/commit/181ab602d532a2a20d64dec68cd712205dbb4ba2#diff-b85f8ec1c03827fdbd450944cf93e466e7e58a438b61f9131ad4e9bf2e1844e6R14)

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html (hunk 6 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/navbar-authenticated.html]: @Squirrel18  Yes:

image

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 9 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: @Squirrel18 yes, in pathways:

image

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 10 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: @Squirrel18 yes:

image

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 11 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: @Squirrel18 yes: 

image

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 12 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: @Squirrel18 yes, we're using it in several sites.

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 13 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: @Squirrel18  No, I'll remove it.

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html (hunk 8 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html]: @Squirrel18  No, I'll remove it.

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html (hunk 15 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: @Squirrel18  Yes, It is.

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html (hunk 17 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: @Squirrel18 I removed it. I added the template from olive and then I added the custom change (PCO_ENABLE_LICENSE_ENFORCEMENT)

File: edx-platform/pearson-vue-theme/lms/templates/ccx/course_name.html (hunk 1 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/ccx/course_name.html]: @Squirrel18 yes, we're using it in several sites.

File: edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html (hunk 16 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/instructor/instructor_dashboard_2/data_download.html]: @Squirrel18 It was present in Juniper Version.

File: edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html (hunk 2 above)
Comment (anfbermudezme) [reply edx-platform/pearson-vue-theme/lms/templates/course_experience/course-dates-fragment.html]: If it is not on the edx platform there is no point in leaving it. @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/courseware/info.html (hunk 3 above)
Comment (anfbermudezme) [reply edx-platform/pearson-vue-theme/lms/templates/courseware/info.html]: Same comment as before. @sergivalero20

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 9 above)
Comment (anfbermudezme) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: We will no longer support courses-pathways.pearson.com.

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+from lms.djangoapps.ccx.overrides import get_current_ccx
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.enterprise_support.utils import get_enterprise_learner_generic_name, get_enterprise_learner_portal
+
+# App that handles subdomain specific branding
+from lms.djangoapps.branding import api as branding_api
+%>
+
+<%
+enterprise_customer_link = get_enterprise_learner_portal(request)
```
Comment (anfbermudezme) [inline edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html]: This module will fail with the new integration of enterprise. Please check if the implementation in juniper stil works.

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

File: edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html (hunk 18 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/navbar-logo-header.html]: @anfbermudezme  I'll remove it

**Review by anfbermudezme — APPROVED** [review]

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html [hunk 19]
```diff
@@ -13,6 +13,7 @@
 from openedx.core.djangoapps.user_api.accounts.toggles import should_redirect_to_order_history_microfrontend
 from openedx.core.djangoapps.user_api.accounts.utils import retrieve_last_sitewide_block_completed
 from openedx.features.enterprise_support.utils import get_enterprise_learner_generic_name, get_enterprise_learner_portal
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
```
Comment (Squirrel18) [inline edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: I think you are not using this import. @sergivalero20

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html (hunk 19 above)
Comment (sergivalero20) [reply edx-platform/pearson-vue-theme/lms/templates/header/user_dropdown.html]: Indeed. I'll remove it.

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**